bayesian_cnn_prometheus/evaluation/bayesian_model_evaluator.py

```diff
@@ -0,0 +1,117 @@
+from copy import copy
+from typing import Tuple, List
+
+import nibabel as nib
+import numpy as np
+from tensorflow.python.keras.engine.training import Model
+from tqdm import tqdm
+
+from bayesian_cnn_prometheus.constants import Paths
+from bayesian_cnn_prometheus.evaluation.utils import load_nifti_file
+from bayesian_cnn_prometheus.learning.model.bayesian_vnet import bayesian_vnet
+
+
+class BayesianModelEvaluator:
+    def __init__(self, weights_path: str, input_shape: Tuple = (32, 32, 16, 1)):
+        """
+        Creates BayesianModelEvaluator object to evaluate bayesian model.
+        :param weights_path: path to bayesian model weights in h5 format
+        :param input_shape: shape of the input to the model
+        """
+        self.input_shape = input_shape
+        self.model = self.load_saved_model(weights_path)
+
+    def evaluate(self, image_path: str, samples_num: int, window: List[int]) -> List[np.array]:
+        """
+        Samples model samples_num times and returns list of predictions.
+        :param image_path: path to the image to predict
+        :param samples_num: number of samples to make
+        :param window: three-elements list with steps value to make in each axis
+        :return: list of arrays with samples_num predictions on image
+        """
+        image = load_nifti_file(image_path)
+        image_chunks, coords = self.create_chunks(image, window)
+
+        predictions = []
+        for _ in tqdm(range(samples_num)):
+            prediction = np.zeros(image.shape)
+            count_prediction = np.zeros(image.shape)
+
+            for chunk, coord in zip(image_chunks, coords):
+                reshaped_chunk = chunk.reshape(1, *chunk.shape, 1)
+                chunk_pred = self.model.predict(reshaped_chunk)
+                reshaped_chunk_pred = chunk_pred.reshape(*chunk.shape)
+
+                prediction = self.add_chunk_to_array(prediction, reshaped_chunk_pred, coord)
+                count_prediction = self.add_chunk_to_array(count_prediction, np.ones(chunk.shape), coord)
+
+            safe_count_prediction = copy(count_prediction)
+            safe_count_prediction[count_prediction == 0] = 1
+
+            predictions.append(prediction / safe_count_prediction)
```

Comment:
Proponuję 
```python
prediction / np.maximum(count_prediction, 1)
```

Reply:
Podmieniam inicjalizacje na `count_prediction = np.ones(image.shape)`

Reply:
To nie jest do końca to samo, bo kiedy zliczasz to zawsze jesteś zbiasowana o 1. 
Problem który widzę jest taki iż będziemy musieli obniżać próg, ponieważ będziemy dostawać mniejsze wartości.
Chociaż nie wiem czy to naprawdę jest problem